packages/wonder-blocks-timing/src/hooks/__tests__/use-timeout.test.ts

```diff
@@ -7,129 +8,484 @@ describe("useTimeout", () => {
         // Arrange
-        const action = jest.fn();
 
         // Act
-        renderHook(() => useTimeout(action, 500, false));
-        jest.advanceTimersByTime(501);
+        const {result} = renderHook(() => useTimeout(null as any, 1000));
 
         // Assert
-        expect(action).not.toHaveBeenCalled();
+        expect(result.error).toEqual(Error("Action must be a function"));
     });
 
-    it("should not fire before the timeout", () => {
+    it("throws if the period is less than 0", () => {
         // Arrange
-        const action = jest.fn();
 
         // Act
-        renderHook(() => useTimeout(action, 500, true));
-        jest.advanceTimersByTime(499);
+        const {result} = renderHook(() => useTimeout(() => {}, -1));
 
         // Assert
-        expect(action).not.toHaveBeenCalled();
+        expect(result.error).toEqual(Error("Timeout period must be >= 0"));
     });
 
-    it("should fire after the timeout", () => {
+    it("should return an ITimeout", () => {
         // Arrange
-        const action = jest.fn();
+        const {result} = renderHook(() => useTimeout(() => {}, 1000));
 
         // Act
-        renderHook(() => useTimeout(action, 500, true));
-        jest.advanceTimersByTime(501);
 
         // Assert
-        expect(action).toHaveBeenCalledTimes(1);
+        expect(result.current).toEqual(
+            expect.objectContaining({
+                clear: expect.any(Function),
+                set: expect.any(Function),
+                isSet: expect.any(Boolean),
+            }),
+        );
     });
 
-    it("should fire after time after 'active' changes to true", () => {
+    it("should default to being immediately set", () => {
         // Arrange
-        const action = jest.fn();
-        const {rerender} = renderHook(
-            ({active}: any) => useTimeout(action, 500, active),
-            {initialProps: {active: false}},
-        );
+        const {result} = renderHook(() => useTimeout(() => {}, 1000));
 
         // Act
-        rerender({active: true});
-        jest.advanceTimersByTime(501);
 
         // Assert
-        expect(action).toHaveBeenCalled();
+        expect(result.current.isSet).toBe(true);
     });
 
-    it("should not fire after the timeout if 'active' changes to false", () => {
-        // Arrange
+    it("should call the action before unmounting", () => {
         const action = jest.fn();
-        const {rerender} = renderHook(
-            ({active}: any) => useTimeout(action, 500, active),
-            {initialProps: {active: true}},
+        const {unmount} = renderHook(() =>
+            useTimeout(action, 1000, {
+                clearPolicy: ClearPolicy.Resolve,
+            }),
         );
 
-        // Act
-        rerender({active: false});
-        jest.advanceTimersByTime(501);
+        act(() => {
+            unmount();
+        });
 
-        // Assert
-        expect(action).not.toHaveBeenCalled();
+        expect(action).toHaveBeenCalled();
     });
```

Comment:
**suggestion (non-blocking):** Arrange/Act/Assert.

Reply:
This was an existing test from your old changes. I didn't edit them all, but the diff has made some unusual choices in showing what changed. I can update them to AAA though.

Reply:
The confusion here is that this is a rename of the old `use-scheduled-timeout.test.ts` file but it's getting diffed against the now deleted `use-timeout.text.ts` file.